homework-2/src/app/components/feature/shows/ReviewForm.tsx

```diff
@@ -0,0 +1,107 @@
+import React, { useState } from "react";
+import {
+  Box,
+  Button,
+  Flex,
+  FormControl,
+  FormLabel,
+  HStack,
+  Icon,
+  Input,
+  Stack,
+  Text,
+} from "@chakra-ui/react";
+
+import { StarIcon } from "@chakra-ui/icons";
+
+interface IReviewFormProps {
+  addShowReview: (review: { rating: number; comment: string }) => void;
+}
+
+export const ReviewForm = ({ addShowReview }: IReviewFormProps) => {
+  const [hoverValue, setHoverValue] = useState<number>(0);
+  const [rating, setRating] = useState<number>(0);
+  const [comment, setComment] = useState<string>("");
+
+  const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    addShowReview({ rating, comment });
+    setRating(0);
+    setComment("");
+  };
+
+  const handleMouseEnter = (index: number) => {
+    setHoverValue(index);
+  };
+
+  const handleMouseLeave = () => {
+    setHoverValue(0);
+  };
+
+  const handleClick = (index: number) => {
+    setRating(index);
+  };
+
+  return (
+    <Box bg=" #3f117c " borderRadius="md" paddingTop="8" paddingBottom="8">
+      <form onSubmit={handleSubmit}>
+        <Stack spacing="4">
+          <FormLabel color="white" fontSize="x-large">
+            Reviews
+          </FormLabel>
+
+          <FormControl>
+            <Input
+              bg="white"
+              type="text"
+              value={comment}
+              onChange={(e) => setComment(e.target.value)}
+              placeholder="Add review"
+            />
+          </FormControl>
+
+          <FormControl>
+            <Box>
+              <Flex alignItems="center" justifyContent="space-bettween">
+                <Text mb="2" color="white">
+                  Rating
+                </Text>
+                <HStack spacing="1">
+                  {Array(5)
```

Comment:
An improvement would be to extract these start in a shared component. That component would keep the hover state and take care of displaying the stars. It would then pass click events to the parent ReviewForm component. It could also be reused on the ReviewItem component.

An additional benefit of this is the following:
As you have implemented it, there are 3 states in the `ReviewForm` component. One of those states is `hoverValue`. Every time you move your mouse over a new star, this state is updated and the whole component is rerendered. If you were to keep this state in a dedicated component, you would rerender only  that component. There is no need to rerender the whole `ReviewForm` (including text input and header) every time someone moves their mouse over the stars.
Remind me to demonstrate this on our meeting today.